### PR TITLE
(PXP-6977): make mds creds required for service startup

### PR DESCRIPTION
**Pull request by johnfrancismccann**
Jira Ticket: [PXP-6977](https://ctds-planx.atlassian.net/browse/PXP-6977)

In combination with the [indexing job updating the Metadata Service](https://github.com/uc-cdis/indexs3client/pull/29) after file upload, require Indexd and MDS creds for all `"name": "indexing"` jobs configured for SSJDispatcher.

This SSJDispatcher PR should go out in the same monthly release as the corresponding [indexing job PR](https://github.com/uc-cdis/indexs3client/pull/29).

### Breaking Changes
- Going forward, to support the [indexing job updating the Metadata Service](https://github.com/uc-cdis/indexs3client/pull/29), SSJDispatcher now requires the Metadata Service to be deployed. Metadata Service creds are now required in the `"imageConfig": {` section for every `"name": "indexing"` job configured for SSJDispatcher. Without both Indexd and Metadata Service creds configured for all `"name": "indexing"` jobs, SSJDispatcher will exit immediately on startup.

### Improvements
- Exit with a status code of 1 instead of 0 when there are errors with reading the credentials file (e.g. it can't be found, no SQS url present, no MDS/Indexd creds)

### Deployment changes
- `metadata-service` must already be present or added to the `"versions": {` section of the `cdis-manifest`. After `metadata-service` has been confirmed to be in the `cdis-manifest`, the now-required Metadata Service creds for SSJDispatcher must be configured by running `gen3 roll all` with `JENKINS_HOME` **not** set. `gen3 roll all` calls `gen3 kube-setup-ssjdispatcher`, which now [automatically configures](https://github.com/uc-cdis/cloud-automation/pull/1428) the SSJDispatcher secret with Metadata Service creds.

### CODE REVIEW

**Review by frickjack — APPROVED** [review]
Awesome!

File: handlers/utils_test.go [hunk 1]
```diff
@@ -68,3 +69,242 @@ func TestUtils(t *testing.T) {
...
+	]
+	`
+	jobConfigs := make([]JobConfig, 0)
+	json.Unmarshal([]byte(jobsJson), &jobConfigs)
+	err := CheckIndexingJobsImageConfig(jobConfigs)
+	assert.Equal(t, err, nil)
+}
+
+// Test that CheckIndexingJobsImageConfig returns an error when MDS creds
+// have not been configured.
```
Comment (paulineribeyre) [inline handlers/utils_test.go:109]: i don't get why we're making this non backwards compatible - what if we need ssjdispatcher in a commons that doesn't use MDS?

Comment (johnfrancismccann) [reply handlers/utils_test.go:109]: This is meant as an extra precaution against the latest SSJDispatcher/indexs3client being deployed to a commons without MDS creds configured for SSJDispatcher, causing this new feature not to work. It's intended to be an automated way to signal to DevOps that SSJDispatcher/indexs3client are not healthy.

If SSJDispatcher is needed in a commons that doesn't use MDS, would it be reasonable to just use an older version of SSJDispatcher?

Also, one thing to keep in mind with this PR is that it only requires MDS creds for `"name": "indexing"` jobs, so it's not going to affect SSJDispatcher deployments in which there is no `indexing` job configured (not sure if that's the case for any commons).

Comment (paulineribeyre) [reply handlers/utils_test.go:109]: > If SSJDispatcher is needed in a commons that doesn't use MDS, would it be reasonable to just use an older version of SSJDispatcher?

no, i don't think so. The ability to upload or index data (SSJDispatcher) is not dependent on whether the MDS is used. MDS is an optional service; all Commons should be able to get the latest SSJDispatcher patches regardless of whether they use it

Comment (giangbui) [reply handlers/utils_test.go:109]: we need to have the backward compatibility. Some commons do not require MDS

Comment (frickjack) [reply handlers/utils_test.go:109]: mds is no longer optional
there's no reason to not run mds
it adds complexity and new ways to eff up if we make it optional

Comment (paulineribeyre) [reply handlers/utils_test.go:109]: then @johnfrancismccann can you update your "Deployment changes" to make it very clear that we need to add the MDS to the manifest

Comment (johnfrancismccann) [reply handlers/utils_test.go:109]: For sure, I'll make note of that in "Deployment changes" in the [indexs3client PR](https://github.com/uc-cdis/indexs3client/pull/29) as well.

**Review by giangbui — APPROVED** [review]
LGTM

**Pull request merged**